gomezmoreno-tp-biblioteca/app.js

```diff
@@ -0,0 +1,253 @@
+        if(nombreLibro != undefined) {
+            libros.push(new Libro(idLibro, nombreLibro, cantidadLibro));
+            res.status(201).json("idLibro: " + idLibro);
+        }
+        else {
+            res.status(400).json("No puede agregarse un libro sin ingresar un titulo");
+        }
+    }
+    else {
+        res.status(400).json("La cantidad debe ser un entero mayor a cero");
+    }
+ })
+
+function libroPrestado(idLibro) {
+    for (var librosPrestamos of prestamos) {
+        if(librosPrestamos.idLibro == idLibro) {
+            return 1;
+        }
+    }
+    return -1;
+}
+
+//RQ1.2: Eliminar libro de la libreria
+app.delete('/libros/:libro', function (req, res) {
+    var eliminarLibro = encontrarIndice(libros, req.params.libro);
+    if(eliminarLibro != -1) {
+        if(libroPrestado(req.params.libro) == -1) {
+            libros.splice(eliminarLibro, 1);
+            res.status(204).send();
+        }
+        else {
+            res.status(400).json("El libro no puede ser eliminado porque está prestado");
+        }
+    }
+    else {
+        res.status(404).json("El id del libro ingresado no existe");
+    }
+})
+
+//RQ1.3: Modificar la cantidad de ejemplares de un libro en la libreria
+app.put('/libros/:libro', function (req, res) {
+    var libroAModificar = encontrarIndice(libros, req.params.libro);
+    var cantidadNueva = req.body.cantidad;
+    if(libroAModificar != -1) {
+        if(Number.isInteger(cantidadNueva) == true && cantidadNueva > 0) {
+            if(cantidadNueva > libros[libroAModificar].disponibles(prestamos)) {
+                libros[libroAModificar].cantidad = cantidadNueva;
+                res.status(204).send();
+            }
+            else {
+                res.status(400).json("No puede ingresarse una cantidad menor a la cantidad de ejemplares prestados");
+            }
+        }
+        else {
+            res.status(400).json("La cantidad debe ser un entero mayor a cero");
+        }
+    }
+    else {
+        res.status(404).json("El id del libro ingresado no existe");
+    }  
+})
+
+//RQ1.4: Obtener la cantidad de ejemplares disponibles para prestamo en la libreria
+app.get('/libros/:libro', function (req, res) {
+    var libroAConsultar = encontrarIndice(libros, req.params.libro);
+    if(libroAConsultar != -1) {
+        res.status(200);
+        var copiaLibro = Object.assign({}, libros[libroAConsultar]);
+        copiaLibro["disponibles"] = copiaLibro.disponibles(prestamos);
+        res.json(copiaLibro);
+    }
+    else {
+        res.status(404).json("El id del libro ingresado no existe");
+    } 
+})
+
+//RQ2.1: Crear socio
+app.post('/socios', function (req, res) {
+    var idSocio = getNewId();
+    var nombreSocio = req.body.nombre;
+    if(nombreSocio != undefined) {
+        socios.push(new Socio(idSocio, nombreSocio));
+        res.status(201).json(idSocio);
+    }
+    else {
+        res.status(400).json("No puede agregarse un socio sin ingresar un nombre");
```

Comment:
use send instead of json

Reply:
I'll use send if it's a 204 code for a delete or a put that I won't get a response message, otherwise I like the json option